maritalk/resources/local.py

```diff
@@ -0,0 +1,157 @@
+import os
+import time
+import shutil
+import atexit
+import subprocess
+from pathlib import Path
+from typing import List, Dict
+import requests
+from requests.exceptions import ConnectionError
+
+
+class MariTalkLocal:
+    def __init__(self, host: str = "localhost", port: int = 8080):
+        self.api_url = f"http://{host}:{port}"
+        self.process = None
+
+    def start_server(
+        self,
+        license: str,
+        port: int = 8080,
+        bin_path: str = f"{Path.home()}/bin/maritalk",
+    ):
+        if not os.path.exists(bin_path):
+            os.makedirs(os.path.dirname(bin_path), exist_ok=True)
+            print("Downloading `maritalk`...")
+            self.download(license, bin_path)
+        print(f"Starting MariTalk Local API at http://localhost:{port}")
+        self.process = subprocess.Popen(
+            [bin_path, "--license", license, "--port", str(port)],
+            stdout=subprocess.DEVNULL,
+            stderr=subprocess.STDOUT,
+        )
+        while True:
+            try:
+                self.status()
+                break
+            except ConnectionError as ex:
+                time.sleep(2)
+
+        def terminate():
+            print("Stopping MariTalk...")
+            self.stop_server()
+
+        atexit.register(terminate)
+
+    def stop_server(self):
+        if not self.process:
+            print("No process attached to this client!")
+            return
+        self.process.terminate()
+        self.process = None
+
+    def download(cls, license: str, bin_path: str):
+        download_url = (
+            "https://m64xplb35dhr3se7ipvtmbdnk40ahktr.lambda-url.us-east-1.on.aws/"
+        )
+        result = requests.post(download_url, json={"license": license})
+        file_url = result.json()["presigned_url"]
+        with requests.get(file_url, stream=True) as response:
+            with open(bin_path, "wb") as out:
+                shutil.copyfileobj(response.raw, out)
+            # Make it executable
+            Path(bin_path).chmod(f.stat().st_mode | stat.S_IEXEC)
```

Comment:
```suggestion
            Path(bin_path).chmod(out.stat().st_mode | stat.S_IEXEC)
```
`f` is undefined

Reply:
Actually, it should't be `out` as well, as I get the following error
AttributeError: '_io.BufferedWriter' object has no attribute 'stat'